File: examples/searchrooms.java
// UCL API example in Java
// This example queries the roombookings/rooms endpoint and prints results.
//
// Owain Kenway

package examples;

import uclapi.UCLApiConnection;
import uclapi.Room;

import java.util.Hashtable;

public class searchrooms {
    public static void main(String[] args) {
        if (args.length == 2) {

            UCLApiConnection conn = new UCLApiConnection(args[0]);
            Hashtable<String, String> params = new Hashtable<String, String>();
            params.put("date", args[1]);

            Room[] rooms = Room.searchAPI(conn, params);

            for (int i = 0; i<rooms.length; i++) {
                System.out.println(rooms[i].toString());
            }
            
        } else {
            System.err.println("Run with API key + date [YYYYMMDD] as arguments.");
            System.exit(1);
        }
    }
}

File: examples/searchrooms.scala
// Person Search example in Scala.
// Owain Kenway

package examples

import uclapi.UCLApiConnection
import uclapi.Room

import java.util.Hashtable

object searchrooms extends App {

    if (args.length != 2) {
        Console.err.println("Run with API key + date [YYYYMMDD] as arguments.")
        System.exit(1)
    }

    val apikey = args(0)
    val params = new Hashtable[String, String]
    params.put("date", args(1))

    val results = Room.searchAPI(new UCLApiConnection(apikey), params)

    for (i <- 0 to (results.length - 1))
        println(results(i).toString())
}